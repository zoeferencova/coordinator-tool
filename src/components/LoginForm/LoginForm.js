import React from 'react';
import AuthApiService from '../../services/auth-api-service'

export default class LoginForm extends React.Component {
    static defaultProps = {
        onLoginSuccess: () => {}
    }

    state = { error: null }

    handleSubmitJwtAuth = e => {
        e.preventDefault()
        this.setState({ error: null })
        const { email, password } = e.target;

        AuthApiService.postLogin({
            email: email.value,
            password: password.value,
        })
            .then(res => {
                email.value = '';
                password.value = '';
                this.props.onLoginSuccess()
            })
            .catch(res => {
                this.setState({ error: res.error })
            })
    }
    
    render() {
        const { error } = this.state;

        return (
            <main>
                <form className='LoginForm' onSubmit={this.handleSubmitJwtAuth}>
                <div role='alert'>{error && <p className='red'>{error}</p>}</div>
                    <div>
                        <label htmlFor="email">Email</label>
                        <input required type="text" name='email' id='login-email' />
                    </div>
                    <div>
                        <label htmlFor="password">Password</label>
                        <input required type="password" name='password' id='login-password' />
                    </div>
                    <button>Cancel</button>
                    <button type='submit'>Log in</button>
                </form>
            </main> 
        )
    }
}