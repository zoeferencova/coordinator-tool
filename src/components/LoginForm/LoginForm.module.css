.formGroup {

    display: flex;
    flex-direction: column;
    width: 70%;
    margin: 30px auto;
}

.label {
    text-align: left;
    font-size: 14px;
    margin-bottom: 10px;
}

.input {
    border-radius: 6px;
    height: 35px;
}

.button {
    font-size: 15px;
    height: 40px;
}

.sign {
    background-color: #507eab;
    color: white;
    font-weight: bold;
    border: none;
}

.sign:hover {
    background-color: #315475;
    border: none;
}