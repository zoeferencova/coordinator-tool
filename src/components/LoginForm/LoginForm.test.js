import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import LoginForm from './LoginForm'

describe(`LoginForm component`, () => {
  it('renders a .LoginForm by default', () => {
    const wrapper = shallow(<LoginForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})