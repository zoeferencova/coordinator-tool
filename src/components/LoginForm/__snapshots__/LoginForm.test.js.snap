// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders a .LoginForm by default 1`] = `
<main>
  <form
    className="LoginForm"
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <div
      className="formGroup"
    >
      <label
        className="label"
        htmlFor="email"
      >
        Email
      </label>
      <Input
        className="input"
        id="login-email"
        name="email"
        required={true}
        type="text"
      />
    </div>
    <div
      className="formGroup"
    >
      <label
        className="label"
        htmlFor="password"
      >
        Password
      </label>
      <Input
        className="input"
        id="login-password"
        name="password"
        required={true}
        type="password"
      />
    </div>
    <Link
      to="/"
    >
      <Button
        className="button cancel"
        type="button"
      >
        Cancel
      </Button>
    </Link>
    <Button
      className="button sign"
      type="submit"
    >
      Sign in
    </Button>
  </form>
</main>
`;
